File: src/main/java/org/kite9/diagram/visualization/planarization/mgt/router/LineRoutingInfo.java
package org.kite9.diagram.visualization.planarization.mgt.router;


/**
 * Line Routing objects are immutable.  They represent a route through the planarization
 */
public interface LineRoutingInfo {
	
	public double getHorizontalRunningCost();
	
	public double getVerticalRunningCost();
	
	public double getRunningCost();
	
}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/mgt/router/LineRoutingInfo.kt
package org.kite9.diagram.visualization.planarization.mgt.router

/**
 * Line Routing objects are immutable.  They represent a route through the planarization
 */
interface LineRoutingInfo {

    fun getHorizontalRunningCost(): Double
    fun getVerticalRunningCost(): Double
    fun getRunningCost(): Double

}